src/webgpu/api/operation/storage_texture/read_only.spec.ts

```diff
@@ -107,6 +107,29 @@ class F extends GPUTest {
                 SetData(texelValue, outputValue, texelDataIndex, component);
                 break;
               }
+              case 'bgra8unorm': {
+                const texelValue = (4 * texelDataIndex + component + 1) % 256;
+                const texelComponentIndex = texelDataIndex * componentCount + component;
+                texelTypedDataView[texelComponentIndex] = texelValue;
+
+                const outputValue = texelValue / 255.0;
+                let outputComponent = 0;
```

Comment:
Further, I thought we can refactor the `SetData` function to
``` typescript
const SetData = (
      texelValue: number,
      outputValue: number,
      texelDataIndex: number,
      component: number,
      outputComponent: number = component,
    ) => {
      const texelComponentIndex = texelDataIndex * componentCount + component;
      texelTypedDataView[texelComponentIndex] = texelValue;
      const outputTexelComponentIndex = texelDataIndex * 4 + outputComponent;
      outputBufferTypedData[outputTexelComponentIndex] = outputValue;
    };
```
and reuse that function.